pacehrh/R/pace_rates_matrix.R

```diff
@@ -36,10 +40,76 @@ generateRatesMatrix <- function(label = NULL){
   return(m)
 }
 
+.getRatesLimits <- function(rateTypeLabel) {
+  limitsTable <- BVE$changeRateLimits
+
+  if (is.null(limitsTable)) {
+    return(c(NA_real_, NA_real_))
+  }
+
+  if (length(grep("Fertility", rateTypeLabel, ignore.case = TRUE)) == 1) {
+    row <- limitsTable[limitsTable$RateCategory == "Fertility", ]
+    limits <- c(row$Min, row$Max)
+  } else if (length(grep("Mortality", rateTypeLabel, ignore.case = TRUE)) == 1) {
+    row <- limitsTable[limitsTable$RateCategory == "Mortality", ]
+    limits <- c(row$Min, row$Max)
+  } else if (tolower(rateTypeLabel) == "incidence") {
+    row <- limitsTable[limitsTable$RateCategory == "Incidence", ]
+    limits <- c(row$Min, row$Max)
+  }
+  else {
+    limits <- c(NA_real_, NA_real_)
+  }
+
+  return(limits)
+}
+
+.getMinMaxRates <- function(initRates, limits) {
+  if (!is.na(limits[1])) {
+    minRates <- initRates * limits[1]
+  } else {
+    minRates <- NULL
+  }
+
+  if (!is.na(limits[2])) {
+    maxRates <- initRates * limits[2]
+  } else {
+    maxRates <- NULL
+  }
+
+  return(list(min = minRates, max = maxRates))
+}
+
+.applyRateLimits <- function(rates, limits) {
+  if (!is.null(limits$max)) {
```

Comment:
there is no checks for max being negative values so this may generate weird results when setting negative max

Reply:
We also need to check if limits$max> limit$min, otherwise the logic does not make sense